### DISCUSSION TITLE
How can I return the sum of two numbers in x86 Intel asm

### DISCUSSION THREAD

**Discussion by ghost** [community/community#66509] [Programming Help]
### Body

‘’’asm 
 section .data
    num1 dd 5
    num2 dd 2
    result dd 0
section .text
    global _start
_start:
    mov eax, num1
    add eax, num2
    syscall
    ‘’’

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ghost** [comment]
Also, what are all the registrars I was bad at computer engineering

**Comment by ghost** [comment]
I am using AMD

**Comment by SreejithMVarma** [comment]
Hi @animeReview,

**Try this:**

```
section .data
    num1 dd 5
    num2 dd 2
    result dd 0

section .text
    global _start

_start:
    ; Load num1 into EAX
    mov eax, [num1]

    ; Add num2 to EAX
    add eax, [num2]

    ; Store the result in the result variable
    mov [result], eax

    ; Exit the program
    mov eax, 1         ; syscall number for sys_exit
    mov ebx, 0         ; exit status: 0
    int 0x80           ; Call kernel

section .bss
    ; Define uninitialized data for result
    resb 4
```
Hope it would help :)

**Comment by comptechNvidia** [comment]
Mark Jp Tolentino

**Comment by comptechNvidia** [comment]
> ### Body
> 
> 
> 
> ‘’’asm 
> 
>  section .data
> 
>     num1 dd 5
> 
>     num2 dd 2
> 
>     result dd 0
> 
> section .text
> 
>     global _start
> 
> _start:
> 
>     mov eax, num1
> 
>     add eax, num2
> 
>     syscall
> 
>     ‘’’
> 
> 
> 
> ### Guidelines
> 
> 
> 
> - [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.
